src/material/autocomplete/testing/autocomplete-harness.ts

```diff
@@ -68,28 +68,33 @@ export class MatAutocompleteHarness extends ComponentHarness {
     return (await this.host()).sendKeys(value);
   }
 
-  /** Gets the autocomplete panel. */
-  async getPanel(): Promise<TestElement> {
-    return this._panel();
-  }
-
   /** Gets the options inside the autocomplete panel. */
-  async getOptions(): Promise<MatAutocompleteOptionHarness[]> {
-    return this._options();
+  async getOptions(filters: OptionHarnessFilters = {}): Promise<MatAutocompleteOptionHarness[]> {
+    return this._documentRootLocator.locatorForAll(MatAutocompleteOptionHarness.with(filters))();
   }
 
   /** Gets the groups of options inside the panel. */
-  async getOptionGroups(): Promise<MatAutocompleteOptionGroupHarness[]> {
-    return this._groups();
+  async getOptionGroups(filters: OptionGroupHarnessFilters = {}):
+      Promise<MatAutocompleteOptionGroupHarness[]> {
+    return this._documentRootLocator.locatorForAll(
+        MatAutocompleteOptionGroupHarness.with(filters))();
   }
 
-  /** Gets whether the autocomplete panel is visible. */
-  async isPanelVisible(): Promise<boolean> {
-    return (await this._panel()).hasClass('mat-autocomplete-visible');
+  /** Selects the first options matching the given filters. */
+  async selectOption(filters: OptionHarnessFilters = {}): Promise<void> {
+    if (!await this.isOpen()) {
+      throw Error('mat-autocomplete dropdown must be open to select an option');
```

Comment:
Can you add another method `openAndSelectOption`?

Reply:
It's a little bit weird because there's no way to guarantee there are any matching options for the current text (and if there are none, the panel won't be opened). What I could do is just change the existing method to ensure that the input is focused first, that way if there are any matching options they'll be displayed

Reply:
It makes sense to me to throw an error if that ends up happening. For a test, that's what you would want anyway. When it's present, it *really* highlights how much code you're saving with the harnesses.

Reply:
Ok, I'll go ahead and roll that functionality into this method. I don't think its worth having one that does the focus and one that doesn't